src/opentelemetry/instrumentation/langchain_v2/callback_handler.py

```diff
@@ -27,19 +40,246 @@ class SpanHolder:
+        return self.span_mapping[run_id].span
+
+    def _end_span(self, span: Span, run_id: UUID) -> None:
+        for child_id in self.span_mapping[run_id].children:
+            child_span = self.span_mapping[child_id].span
+            if child_span.end_time is None:  # avoid warning on ended spans
+                child_span.end()
+        span.end()
         
+    def _create_span(
+            self,
+            run_id: UUID,
+            parent_run_id: Optional[UUID],
+            span_name: str,
+            kind: SpanKind = SpanKind.INTERNAL,
+            workflow_name: str = "",
+            entity_name: str = "",
+            entity_path: str = "",
+            metadata: Optional[dict[str, Any]] = None,
+        ) -> Span:
+            if metadata is not None:
+                current_association_properties = (
+                    context_api.get_value("association_properties") or {}
+                )
+                sanitized_metadata = {
+                    k: _sanitize_metadata_value(v)
+                    for k, v in metadata.items()
+                    if v is not None
+                }
+                context_api.attach(
+                    context_api.set_value(
+                        "association_properties",
+                        {**current_association_properties, **sanitized_metadata},
+                    )
+                )
+
+            if parent_run_id is not None and parent_run_id in self.span_mapping:
+                span = self.tracer.start_span(
+                    span_name,
+                    context=set_span_in_context(self.span_mapping[parent_run_id].span),
+                    kind=kind,
+                )
+            else:
+                span = self.tracer.start_span(span_name, kind=kind)
+
+            # ////////// below 2 are not OTel
+            # _set_span_attribute(span, SpanAttributes.TRACELOOP_WORKFLOW_NAME, workflow_name)
+            # _set_span_attribute(span, SpanAttributes.TRACELOOP_ENTITY_PATH, entity_path)
+
+            # ////////// not OTel, cant find usage so potentially remove later
+            token = context_api.attach(
+                context_api.set_value(SUPPRESS_LANGUAGE_MODEL_INSTRUMENTATION_KEY, True)
+            )
+
+            self.span_mapping[run_id] = SpanHolder(
+                span, token, None, [], workflow_name, entity_name, entity_path
+            )
+
+            if parent_run_id is not None and parent_run_id in self.span_mapping:
+                self.span_mapping[parent_run_id].children.append(run_id)
+
+            return span
+
+        
+        
+    def _create_llm_span(
+        self,
+        run_id: UUID,
+        parent_run_id: Optional[UUID],
+        name: str,
+        operation_name: GenAIOperationValues,
+        metadata: Optional[dict[str, Any]] = None,
+    ) -> Span:
+        workflow_name = self.get_workflow_name(parent_run_id) 
+        entity_path = self.get_entity_path(parent_run_id) 
```

Comment:
It doesn't look like these are used outside of TraceLoop-defined span attributes, can probably remove